### DISCUSSION TITLE
Text to speech for scientific PDF files

### DISCUSSION THREAD

**Discussion by codingbutstillalive** [omnivore-app/omnivore#2132] [Ideas]
I imagine this to be tricky, but quite useful.

**Comment by jacksonh** [comment]
Yes, this one will be hard. We plan on changing our PDF library over the summer, maybe once we do that it will be a bit easier to handle, however its very challenging to find the order of text in a PDF vs a web article

**Reply by papatistos** [reply]
While I would really like this feature, I think it should probably not be a priority because it is not part of the core omnivore functionality. There are apps specialized on TTS for PDFs and even those are mostly doing a bad job. From that, I conclude that this would be a black hole for omnivore resources. 

Figuring out the order in which the text should be read is the smallest of challenges and has, I believe, been more or less solved. The tricky part is cropping out headers, footers, and diagrams in a reliable way (even doing it manually often fails when the position of the text in the image doesn’t align with the invisible underlying text). And then you need to figure out ways to exclude references and footnote numbers from being read out. 

Personally, I have landed on [TTS pdf reader](https://apps.apple.com/app/apple-store/id1519637964?pt=96241807&ct=christoph&mt=8), which works best and has high-quality voices that I can listen to for hours without problems. Actually, in English, it’s one voice in particular that is really good: Amy (neural).

I’m absolutely not saying that omnivore should not do pdf tts, I’m just pointing out that it’s a can of worms to open.

Before opening that, I would prefer if TTS for webpages could be made available on desktop :-)

**Reply by paulrudy** [reply]
Just jumping in to say that [Voice Dream Reader](https://www.voicedream.com) does a good job with TTS on PDFs as well. It would be awesome to have available in Omnivore.

See also #2356
